### PR TITLE
convert as+null check to is pattern match

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: src/NuGetGallery.Core/Infrastructure/TableErrorLog.cs [hunk 1]
```diff
@@ -195,8 +195,7 @@ public static void Obfuscate(Error error)
             }
 
             //ServerVariables overrides requiring context from the http request should be handled in NuGetGallery.QuietLog
-            var elmahException = error.Exception as ElmahException;
-            if (elmahException != null)
+            if (error.Exception is ElmahException elmahException)
```
Comment (erdembayar) [inline src/NuGetGallery.Core/Infrastructure/TableErrorLog.cs:198]: This file going to be deleted in another PR.

Comment (SimonCropp) [reply src/NuGetGallery.Core/Infrastructure/TableErrorLog.cs:198]: thats fine. i will handle the conflict once it is deleted

**Comment by erdembayar** [comment]
@SimonCropp 
There is merge conflict in this PR.

**Comment by SimonCropp** [comment]
resolved

**Comment by SimonCropp** [comment]
second round of conflicts resolved. can we merge this one?
